src/test/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketWebhookClientImplTest.java

```diff
@@ -0,0 +1,87 @@
+package com.atlassian.bitbucket.jenkins.internal.client;
+
+import com.atlassian.bitbucket.jenkins.internal.fixture.FakeRemoteHttpServer;
+import com.atlassian.bitbucket.jenkins.internal.http.HttpRequestExecutorImpl;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketWebhook;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.atlassian.bitbucket.jenkins.internal.client.BitbucketCredentials.ANONYMOUS_CREDENTIALS;
+import static com.atlassian.bitbucket.jenkins.internal.util.TestUtils.*;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toSet;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.IsIterableContaining.hasItems;
+import static org.junit.Assert.assertThat;
+
+public class BitbucketWebhookClientImplTest {
```

Comment:
Please add integration tests (`BitbucketWebhookClientIT `).  Have a look at `BitbucketSearchEndpointIT` for inspiration.

Reply:
I will add it as a part of different PR. Reason being, I would like to include as many classes as possible and there are more classes to be added.

Reply:
Could you please raise an issue for tracking this work?